AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -542,6 +533,26 @@ - (void)resumeWithIdentifyingObjectSync:(id <NSObject>)identifyingObject {
                             }];
 }
 
+- (void)pauseSendingLogsWithToken:(NSString *)token {
+  NSString *targetKey = [MSUtility targetKeyFromTargetToken:token];
+  @synchronized (self.pausedTargetKeys) {
+    MSLogDebug([MSAppCenter logTag], @"Pause channel for target key %@.", targetKey);
+    [self.pausedTargetKeys addObject:targetKey];
+  }
+}
+
+- (void)resumeSendingLogsWithToken:(NSString *)token {
+  NSString *targetKey = [MSUtility targetKeyFromTargetToken:token];
+  @synchronized (self.pausedTargetKeys) {
+    MSLogDebug([MSAppCenter logTag], @"Resume channel for target key %@.", targetKey);
+    [self.pausedTargetKeys removeObject:targetKey];
+
+    // Update item count and check logs if it meets the conditions to send logs.
+    self.itemsCount = [self.storage countLogs];
```

Comment:
Doing the count now works but ideally we should refresh the count every-time  we call `checkPendingLogs`. Basically querying for count within `checkPendingLogs` instead of the letting the channel doing the counting itself line 199. That way we can use `checkPendingLogs` everywhere instead of flush.